ete-test/ete.py

```diff
@@ -0,0 +1,336 @@
+	assert len(flags_data) > 0
+
+#***********************************************************************************************************#
+#ACTUAL TESTS###################################################################################ACTUAL TESTS#
+#############################################################################################################
+
+#Test first (input) row: contains all None's, except date.
+#Row_id can't be null because it's automatically incremented
+def test_row_1(flagged_data, flags_data):
+	#Step 1: Need to split flags to have only _NULL flags
+	mask = flags_data.description.str.contains("_NULL")
+	null_flags = flags_data[mask]
+
+	#Should have more than 1 NULL flag
+	assert len(null_flags) > 1
+
+	#Step 2: Need to split flagged to contain data for Row 1 only 
+	mask = flagged_data.row_id == 1
+	first_row = flagged_data[mask]
+
+	#Should have more than flagged 1 row for first inout row
+	assert len(first_row) > 1
+
+	#Step 3: Iterate over null flags omiting service_date
+	for index, row in null_flags.iterrows():
+		if not ((row["description"] == "SERVICE_DATE_NULL") or (row["description"] == "ROW_ID_NULL")):
+			#for each null_flag that we want to check: check if first row contains it (must contain all null flags except 3 above)
+			assert any(first_row["flag_id"] == row["flag_id"]) == True
+
+#Test second (input) row: all data is good, except there is data for unobserved_stop flag to be turned on
+def test_row_2(flagged_data, flags_data):
+	#Step 1: Need to split flags to have only UNOBSERVED_STOP flag
+	mask = flags_data.description.str.contains("UNOBSERVED_STOP")
+	unobserved_flag_row = flags_data[mask]
+	unobserved_flag = unobserved_flag_row["flag_id"].values[0]
+
+	#Must only have 1 row
+	assert len(unobserved_flag_row) == 1
+
+	#Step 2: Only need data for 2nd row
+	mask = flagged_data.row_id == 2
+	second_row = flagged_data[mask]
+	second_row_flag = second_row["flag_id"].values[0]
+
+	#Must only have 1 row
+	assert len(second_row) == 1
+
+	#Step 3: Confirm flag is turned on
+	assert unobserved_flag == second_row_flag
+
+
+#Test third (input) row: all data is good, except there is data for unopened_door flag to be turned on
+def test_row_3(flagged_data, flags_data):
+	#Step 1: Need to split flags to have only UNOBSERVED_STOP flag
+	mask = flags_data.description.str.contains("UNOPENED_DOOR")
+	unopened_door_flag_row = flags_data[mask]
+	unopened_door_flag = unopened_door_flag_row["flag_id"].values[0]
+
+	#Must only have 1 row
+	assert len(unopened_door_flag_row) == 1
+
+	#Step 2: Only need data for 3rd row
+	mask = flagged_data.row_id == 3
+	third_row = flagged_data[mask]
+	third_row_flag = third_row["flag_id"].values[0]
+
+	#Must only have 1 row
+	assert len(third_row) == 1
+
+	#Step 3: Confirm flag is turned on
+	assert unopened_door_flag == third_row_flag
+
+'''
+#Test fourth and fifth (rows): all data is good, but they're duplicates of each other, therefore both must contain duplicate flag
```

Comment:
Why is this one gaint string?

Reply:
""" and ''' are Python's semi-equivalent of multi-lline comments.

Reply:
Do you mean comment or commented out code?

Reply:
Why do we have this test commented out? If we're not using it it should be removed

Reply:
Duplicate flagger is not implemented correctly (Sawyer is working on it I believe), which fails these 2 tests, along with test #2 (adds additional flags, thus throwing off correct count)